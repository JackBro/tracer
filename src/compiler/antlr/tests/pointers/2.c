void main(){
	int *x, y, a[5];

	(*x) = y + 2;

	(a[2]) = 3;

	(*(x+1)) = (*x) + y;

	(a[*x]) = ((a[y]) + 4);
}

