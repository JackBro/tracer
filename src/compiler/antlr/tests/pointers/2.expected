trans(func_main, i5,
	[p$init_0],[H$1$,H$2$,H$3$,H$4$],[],[SH$1$,SH$2$,SH$3$,SH$4$],[],[A],[ReturnValue,X,Y],[SA],[SReturnValue,SX,SY],
	[p$init_1],[H$1$,H$2$,H$3$,H$4$],[],[SH$1$,SH$2$,SH$3$,SH$4$],[],[A],[ReturnValue,X,Y],[SA],[SReturnValue,SX,SY],
	[],
	[],
	[],
	not_branch).
trans(func_main, i6,
	[p$init_1],[H$1$,H$2$,H$3$,H$4$],[],[SH$1$,SH$2$,SH$3$,SH$4$],[],[A],[ReturnValue,X,Y],[SA],[SReturnValue,SX,SY],
	[p0],[H$1$,H$2$,H$3$,H$4$],[],[SH$1$,SH$2$,SH$3$,SH$4$],[],[A],[ReturnValue,X,Y],[SA],[SReturnValue,SX,SY],
	[],
	[call(func___TRACER_INIT,[RV0|_],[SRV0|_],[])],
	[],
	not_branch).
trans(func_main, i1,
	[p0],[H$1$,H$2$,H$3$,H$4$],[],[SH$1$,SH$2$,SH$3$,SH$4$],[],[A],[ReturnValue,X,Y],[SA],[SReturnValue,SX,SY],
	[p7],[H$1$,H$2$,H$3$,H$4$$Next$],[],[SH$1$,SH$2$,SH$3$,SH$4$$Next$],[],[A$Next$],[ReturnValue,X,Y],[SA$Next$],[SReturnValue,SX,SY],
	[arrassign(SH$4$$1$,upd(SH$4$,SX,plus(SY,2))),arrassign(SA$1$,upd(SA,2,3)),arrassign(SH$4$$Next$,upd(SH$4$$1$,plus(SX,1),plus(ref(SH$4$$1$,SX),SY))),arrassign(SA$Next$,upd(SA$1$,ref(SH$4$$Next$,SX),plus(ref(SA$1$,SY),4)))],
	[],
	[ref(H$4$,X,H$4$Ref0),ref(A,2,ARef0),ref(H$4$$1$,(X+1),H$4$$1$Ref0),ref(H$4$$1$,X,H$4$$1$Ref1),ref(A$1$,H$4$$Next$Ref0,A$1$Ref0),ref(H$4$$Next$,X,H$4$$Next$Ref0),ref(A$1$,Y,A$1$Ref1)],
	not_branch) :-
	H$4$$1$=upd(H$4$,X,Y+2),A$1$=upd(A,2,3),H$4$$Next$=upd(H$4$$1$,(X+1),(H$4$$1$Ref1)+Y),A$Next$=upd(A$1$,H$4$$Next$Ref0,((A$1$Ref1)+4)).
trans(func_main, i2,
	[p7],[H$1$,H$2$,H$3$,H$4$],[],[SH$1$,SH$2$,SH$3$,SH$4$],[],[A],[ReturnValue,X,Y],[SA],[SReturnValue,SX,SY],
	[p8],[H$1$,H$2$,H$3$,H$4$],[],[SH$1$,SH$2$,SH$3$,SH$4$],[],[A],[ReturnValue$Next$,X,Y],[SA],[SReturnValue$Next$,SX,SY],
	[assign(SReturnValue$Next$,_)],
	[],
	[],
	not_branch) :-
	ReturnValue$Next$=_.
trans(func___TRACER_INIT, i3,
	[p0],[H$1$,H$2$,H$3$,H$4$],[],[SH$1$,SH$2$,SH$3$,SH$4$],[],[],[ReturnValue],[],[SReturnValue],
	[p1],[H$1$,H$2$,H$3$,H$4$],[],[SH$1$,SH$2$,SH$3$,SH$4$],[],[],[ReturnValue],[],[SReturnValue],
	[],
	[],
	[],
	not_branch).
trans(func___TRACER_INIT, i4,
	[p1],[H$1$,H$2$,H$3$,H$4$],[],[SH$1$,SH$2$,SH$3$,SH$4$],[],[],[ReturnValue],[],[SReturnValue],
	[p2],[H$1$,H$2$,H$3$,H$4$],[],[SH$1$,SH$2$,SH$3$,SH$4$],[],[],[ReturnValue$Next$],[],[SReturnValue$Next$],
	[assign(SReturnValue$Next$,_)],
	[],
	[],
	not_branch) :-
	ReturnValue$Next$=_.
%--------------------------------------------------
block(b1, func_main, [_], [_], _, nl, [i1,i2,i5,i6],  [n,n,n,u],[],[u],[u,n,n]).
block(b2, func___TRACER_INIT, [_], [_], _, nl, [i3,i4],  [n,n,n,n],[],[],[u]).
%--------------------------------------------------
start(func_main, [p$init_0]).
start(func___TRACER_INIT, [p0]).
%--------------------------------------------------
bottom(func_main, [p8]).
bottom(func___TRACER_INIT, [p2]).
%--------------------------------------------------
freshglobalarrays([SH$1$,SH$2$,SH$3$,SH$4$]).
%--------------------------------------------------
freshglobalvars([]).
%--------------------------------------------------
freshlocalarrays(func_main,[SA]).
freshlocalarrays(func___TRACER_INIT,[]).
%--------------------------------------------------
freshlocalvars(func_main,[SReturnValue,SX,SY]).
freshlocalvars(func___TRACER_INIT,[SReturnValue]).
%--------------------------------------------------
freshparams(func_main,[lv(0)]).
freshparams(func___TRACER_INIT,[lv(0)]).
%--------------------------------------------------
init_goal(func_main, [p$init_0], i5, [H$1$,H$2$,H$3$,H$4$],[],[SH$1$,SH$2$,SH$3$,SH$4$],[],[A],[ReturnValue,X,Y],[SA],[SReturnValue,SX,SY],[]).
